src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+            self.did_info.parent_info,
+            inner_puzzle,
+        )
+        await self.save_info(new_info)
+
+        future_parent = CCParent(
+            coin.parent_coin_info,
+            inner_puzzle.get_tree_hash(),
+            coin.amount,
+        )
+
+        await self.add_parent(coin.name(), future_parent)
+
+        for name, ccparent in self.did_info.parent_info:
+            if coin.parent_coin_info == name:
+                search_for_parent = False
+                break
+
+        if search_for_parent:
+            data: Dict[str, Any] = {
+                "data": {
+                    "action_data": {
+                        "api_name": "request_generator",
+                        "height": height,
+                        "header_hash": header_hash,
+                    }
+                }
+            }
+
+            data_str = dict_to_json_str(data)
+            await self.wallet_state_manager.create_action(
+                name="request_generator",
+                wallet_id=self.wallet_info.id,
+                type=self.wallet_info.type,
+                callback="generator_received",
+                done=False,
+                data=data_str,
+            )
+
+    # This should basically never be called as we don't want to receive ID coins from somebody else
+    async def search_for_parent_info(
+        self, block_program: Program, removals: List[Coin]
+    ) -> bool:
+
+        """
+        Returns an error if it's unable to evaluate, otherwise
+        returns a list of NPC (coin_name, solved_puzzle_hash, conditions_dict)
+        """
+        cost_sum = 0
+        try:
+            cost_run, sexp = block_program.run_with_cost([])
+            cost_sum += cost_run
+        except EvalError:
+            return False
+
+        for name_solution in sexp.as_iter():
+            _ = name_solution.as_python()
+            if len(_) != 2:
+                return False
+            if not isinstance(_[0], bytes) or len(_[0]) != 32:
+                return False
+            coin_name = bytes32(_[0])
+            if not isinstance(_[1], list) or len(_[1]) != 2:
+                return False
+            puzzle_solution_program = name_solution.rest().first()
+            puzzle_program = puzzle_solution_program.first()
+            try:
+                error, conditions_dict, cost_run = conditions_dict_for_solution(
+                    puzzle_solution_program
+                )
+                cost_sum += cost_run
+                if error:
+                    return False
+            except clvm.EvalError:
+
+                return False
+            if conditions_dict is None:
+                conditions_dict = {}
+
+            if ConditionOpcode.CREATE_COIN in conditions_dict:
+                created_output_conditions = conditions_dict[ConditionOpcode.CREATE_COIN]
+            else:
+                continue
+            for cvp in created_output_conditions:
```

Comment:
There must be some refactored code somewhere that gets a list of output conditions for a puzzle's solution.